src/value.rs

```diff
@@ -42,46 +42,86 @@ unsafe impl FromValue for Value {
     }
 }
 
-const NONE: Value = Value(sys::NONE);
-const UNIT: Value = Value(sys::UNIT);
+unsafe impl<'a, T> IntoValue for OCaml<'a, T> {
+    fn into_value(self, _rt: &mut Runtime) -> Value {
+        unsafe { Value::new(self.raw()) }
+    }
+}
+
+unsafe impl<'a, T> FromValue for OCaml<'a, T> {
+    fn from_value(value: Value) -> OCaml<'a, T> {
+        let rt = unsafe { &mut Runtime::recover_handle() };
+        let x: OCaml<T> = unsafe { OCaml::new(rt, value.0) };
+        unsafe { core::mem::transmute(x) }
```

Comment:
Same as the `OCamlRef` conversion. I think that by default you can assume that for any case where you are using `Runtime::recover_handle()`, the conversion you are making is not safe.